### PR TITLE
Create auto-bicep.yml

### PR DESCRIPTION
**Pull request by dciborow**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: .github/workflows/auto-bicep.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: Auto Build Bicep
+
+on:
+  push:
+    branches-ignore:
+      - master
+    paths:
```
Comment (bmoore-msft) [inline .github/workflows/auto-bicep.yml:7]: I think we should exclude the ```test``` folder

File: .github/workflows/auto-bicep.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          TESTFOLDER_PATH=$SAMPLEFOLDER_PATH
+          FOUNDFOLDER_PATH=
+          while [ "$TESTFOLDER_PATH" != "." ]
+          do
+            echo "Looking for main template in $TESTFOLDER_PATH"
+            MAINBICEP_PATH=$TESTFOLDER_PATH/main.bicep
+            AZDEPLOYJSON_PATH=$TESTFOLDER_PATH/azuredeploy.json
+            if [ -f "$MAINBICEP_PATH" ] || [ -f "$AZDEPLOYJSON_PATH" ]; then
+              FOUNDFOLDER_PATH=$TESTFOLDER_PATH
+              echo Found main template in $FOUNDFOLDER_PATH
```
Comment (bmoore-msft) [inline .github/workflows/auto-bicep.yml]: what if there is more than one?

File: .github/workflows/auto-bicep.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          echo "AZDEPLOYJSON_PATH=$AZDEPLOYJSON_PATH" >> $GITHUB_ENV
+
+      - name: Build main.bicep -> azuredeploy.json
+        uses: Azure/cli@v1
+        with:
+          # Specify the script here
+          inlineScript: |
+            if [ -f $MAINBICEP_PATH ]; then          
+              echo Running: az bicep build --file $MAINBICEP_PATH --outfile $AZDEPLOYJSON_PATH
+              az bicep build --file $MAINBICEP_PATH --outfile $AZDEPLOYJSON_PATH
```
Comment (bmoore-msft) [inline .github/workflows/auto-bicep.yml]: we probably need to make sure we have the latest version - the other pipelines use that so if we don't here the bicep build in the main pipeline will not match this one

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-bicep.yml (hunk 1 above)
Comment (dciborow) [reply .github/workflows/auto-bicep.yml:7]:
```suggestion
    paths:
      - 'tests/**'
```

File: .github/workflows/auto-bicep.yml (hunk 3 above)
Comment (dciborow) [reply .github/workflows/auto-bicep.yml]: i believe that i am getting latest version... but I am going to add something to echo the versions so they are in the output.

**Pull request merged**